File: JAVA/rental-app-java/src/utils/DateUtils.java
package utils;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeParseException;

public class DateUtils {
    public static LocalDate parseDate(String dateStr) {
        try {
            return LocalDate.parse(dateStr);
        } catch (DateTimeParseException e) {
            return null;
        }
    }

    public static boolean isValidRange(LocalDate start, LocalDate end) {
        return start != null && end != null && !end.isBefore(start.plusDays(1));
    }
}

File: JS/rental-app-js/utils/dateUtils.js
function parseDate(dateStr) {
    const date = new Date(dateStr);
    return isNaN(date.getTime()) ? null : date;
  }
  
  function isValidRange(start, end) {
    if (!start || !end) return false;
    const oneDay = 24 * 60 * 60 * 1000;
    return end - start >= oneDay;
  }
  
  module.exports = { parseDate, isValidRange };
  